### DISCUSSION TITLE
Check for missing argument or incorrect argument data type in call to function 'is'.  Error in optimizer (line 105)         if is(u{i},'complex')

### DISCUSSION THREAD

**Discussion by nbahari** [yalmip/YALMIP#1102] [Problems using YALMIP] [answered]
Hello All, 
I'm working on a fairly big toolbox and I'm using YALMiP with GUROBI. I keep facing an error whose source of error I can not narrow down to. This error has kept popping up.   I kept trying to figure out what the problem could be. I wish I could be able to provide a piece of code where I assumed to be the problem source. I checked if there is any imaginary number in the inputs using the function `~isreal`. There is no imaginary number in the inputs.

My question is if there are some common causes for this error. The error is:

```
Check for missing argument or incorrect argument data
type in call to function 'is'.

Error in optimizer (line 105)
        if is(u{i},'complex')
```
The whole error message I get is the following. This shows there is something wrong in the input arguments of the optimizer that I can not figure out.

```
Check for missing argument or incorrect argument data
type in call to function 'is'.

Error in optimizer (line 105)
        if is(u{i},'complex')

Error in STLC_get_controller (line 375)
controller = optimizer([ Fs,Fdyn,Fstl,Fr_stl,
Fu],[],options,param_controller, output_controller);
```
Any help is greatly appreciated!

**Comment by johanlofberg** [comment, marked as answer]
Your output_controller argument is not an sdpvar

```
>> is(sdpvar,'complex')

ans =

  logical

   0

>> is([],'complex')
Check for missing argument or
incorrect argument data type in
call to function 'is'.
 
>> is(0,'complex')
Check for missing argument or
incorrect argument data type in
call to function 'is'.
```

**Reply by nbahari** [reply]
Thank you for your reply!
